src/reactviews/pages/ExecutionPlan/highlightExpensiveOperations.tsx

```diff
@@ -0,0 +1,176 @@
+		maxHeight: "200px"
+	}
+})
+
+interface HighlightExpensiveOperationsProps {
+	executionPlanView: any;
+	setExecutionPlanView: any;
+	setHighlightOpsClicked: any;
+}
+
+export const HighlightExpensiveOperations: React.FC<HighlightExpensiveOperationsProps> = ({
+		executionPlanView,
+		setExecutionPlanView,
+		setHighlightOpsClicked
+  	}) => {
+	const classes = useStyles();
+	const state = useContext(ExecutionPlanContext);
+	const executionPlanState = state?.state;
+	const LocalizedConstants = executionPlanState!.localizedConstants!;
+	const [highlightMetricSelected, setHighlightMetricSelected] = useState('');
+
+	const highlightMetricOptions: string[] =
+		[LocalizedConstants.actualElapsedTime, LocalizedConstants.actualElapsedCpuTime, LocalizedConstants.cost, LocalizedConstants.subtreeCost,
+			LocalizedConstants.actualNumberOfRows, LocalizedConstants.numRowsRead, LocalizedConstants.off];
+	const highlightMetricOptionsEnum: ep.ExpensiveMetricType[] =
+	 [ep.ExpensiveMetricType.ActualElapsedTime, ep.ExpensiveMetricType.ActualElapsedCpuTime, ep.ExpensiveMetricType.Cost, ep.ExpensiveMetricType.SubtreeCost, ep.ExpensiveMetricType.ActualNumberOfRowsForAllExecutions, ep.ExpensiveMetricType.NumberOfRowsRead, ep.ExpensiveMetricType.Off];
+
+
+	const handleHighlightExpensiveOperation = async () => {
+		if (executionPlanView) {
+			const enumSelected = highlightMetricOptionsEnum[highlightMetricOptions.indexOf(highlightMetricSelected)];
+			const expensiveOperationDelegate: (cell: ep.AzDataGraphCell) => number | undefined = getExpensiveOperationDelegate(enumSelected)!;
+			executionPlanView.clearExpensiveOperatorHighlighting();
+			const elementId = executionPlanView.highlightExpensiveOperator(expensiveOperationDelegate);
+			if (elementId) {
+				executionPlanView.centerElement(executionPlanView.getElementById(elementId)!);
+			}
+			setExecutionPlanView(executionPlanView);
+		}
+	};
+
+	const handleHighlightClose = async () => {
+		if (executionPlanView) {
+			executionPlanView.clearExpensiveOperatorHighlighting();
+			setExecutionPlanView(executionPlanView);
+		}
+		setHighlightOpsClicked(false);
+	};
+
+	return (
+		<div id="highlightExpensiveOpsContainer" className={classes.inputContainer} style={{background:utils.iconBackground(executionPlanState!.theme!)}}>
+			<div>{LocalizedConstants.metric}</div>
+			<Combobox id="highlightExpensiveOpsDropdown" onOptionSelect={(_, data) => setHighlightMetricSelected(data.optionText ?? '')}>
+				<div style={{maxHeight:"250px"}}>
+				{highlightMetricOptions.map((option) => (
+					<Option key={option}>
+						{option}
+					</Option>
+				))}
+				</div>
+			</Combobox>
+			<Button onClick={handleHighlightExpensiveOperation} icon={<Checkmark20Regular />} />
+			<Button icon={<Dismiss20Regular />} onClick={handleHighlightClose}/>
+		</div>
+
+	);
+};
+
+
+function getExpensiveOperationDelegate(selectedExpensiveOperationType: ep.ExpensiveMetricType): (cell: ep.AzDataGraphCell) => number | undefined {

```

Comment:
I am guessing you picked these functions from ads. Maybe we can move them to the azdatagraph to prevent duplication in the next iteration.